qt/scientific_interfaces/Indirect/IndirectFitData.h

```diff
@@ -0,0 +1,161 @@
+#ifndef MANTIDQTCUSTOMINTERFACESIDA_INDIRECTFITDATA_H_
+#define MANTIDQTCUSTOMINTERFACESIDA_INDIRECTFITDATA_H_
+
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidKernel/ArrayProperty.h"
+
+#include <boost/optional.hpp>
+#include <boost/variant.hpp>
+#include <boost/variant/static_visitor.hpp>
+#include <boost/weak_ptr.hpp>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace IDA {
+
+template <typename T>
+std::vector<T> vectorFromString(const std::string &listString) {
+  try {
+    return Mantid::Kernel::ArrayProperty<T>("vector", listString);
+  } catch (const std::runtime_error &) {
+    return std::vector<T>();
+  }
+}
+
+template <typename T> class DiscontinuousSpectra {
+public:
+  explicit DiscontinuousSpectra(const std::string &str)
+      : m_str(str), m_vec(vectorFromString<T>(str)) {}
+  DiscontinuousSpectra(const DiscontinuousSpectra &vec)
+      : m_str(vec.m_str), m_vec(vec.m_vec) {}
+  DiscontinuousSpectra(DiscontinuousSpectra &&vec)
+      : m_str(std::move(vec.m_str)), m_vec(std::move(vec.m_vec)) {}
+
+  DiscontinuousSpectra &operator=(const DiscontinuousSpectra &vec) {
+    m_str = vec.m_str;
+    m_vec = vec.m_vec;
+    return *this;
+  }
+
+  DiscontinuousSpectra &operator=(DiscontinuousSpectra &&vec) {
+    m_str = std::move(vec.m_str);
+    m_vec = std::move(vec.m_vec);
+    return *this;
+  }
+
+  bool empty() const { return m_vec.empty(); }
+  std::size_t size() const { return m_vec.size(); }
+  const std::string &getString() const { return m_str; }
+  typename std::vector<T>::iterator begin() { return m_vec.begin(); }
+  typename std::vector<T>::iterator end() { return m_vec.end(); }
+  typename std::vector<T>::const_iterator begin() const {
+    return m_vec.begin();
+  }
+  typename std::vector<T>::const_iterator end() const { return m_vec.end(); }
+  const T &operator[](std::size_t index) const { return m_vec[index]; }
+
+private:
+  std::string m_str;
+  std::vector<T> m_vec;
+};
+
+using Spectra = boost::variant<DiscontinuousSpectra<std::size_t>,
```

Comment:
The variant here allows for two representations of spectra as the fastest operations performed on a discontinuous range (equivalent to a vector of spectrum) or a continuous range (equivalent to a pair defining the bounds) are not necessarily the same. There are circumstances in which it is more optimal to represent spectra as a pair (e.g. checking for out of bounds input, combining two spectra selections, etc.).

The discontinuous spectra is a template class in case of the fitting model being used in other contexts, where spectrum may be represented as a plain int or other type. It's not particularly necessary in this context.